examples/roitberg/roitberg_ANI.py

```diff
@@ -0,0 +1,222 @@
+
+            yield tmp_dataset.X, tmp_dataset.y, tmp_dataset.w, tmp_dataset.ids
+
+            X_cache = []
+            y_cache = []
+            w_cache = []
+            ids_cache = []
+
+          else:
+            X_cache.append(X)
+            y_cache.append(y)
+            w_cache.append(1)
+            ids_cache.append(row_idx)
+            groups.append(group_idx)
+
+          row_idx += 1
+
+        group_idx += 1
+
+    # flush once more at the end
+    if len(X_cache) > 0:
+      tmp_dataset = dc.data.NumpyDataset(
+          np.array(X_cache),
+          np.array(y_cache), np.array(w_cache), np.array(ids_cache))
+
+      tmp_dataset = ANItransformer.transform(tmp_dataset)
+
+      yield tmp_dataset.X, tmp_dataset.y, tmp_dataset.w, tmp_dataset.ids
+
+  tasks = ["ani"]
+  dataset = dc.data.DiskDataset.create_dataset(shard_generator(), tasks=tasks)
+  ANItransformer.sess.close()
+
+  print("Number of groups", np.amax(groups))
+  print("dataset_shape", dataset.X.shape)
+  splitter = dc.splits.RandomGroupSplitter(groups)
+  train_dataset, valid_dataset, test_dataset = splitter.train_valid_test_split(
+      dataset,
+      frac_train=.5,
+      frac_valid=.25,
+      frac_test=.25,)
+
+  transformers = [
+      dc.trans.NormalizationTransformer(
+          transform_y=True, dataset=train_dataset)
+  ]
+
+  for transformer in transformers:
+    train_dataset = transformer.transform(train_dataset)
+    valid_dataset = transformer.transform(valid_dataset)
+    test_dataset = transformer.transform(test_dataset)
+
+  return train_dataset, valid_dataset, test_dataset, transformers
+
+
+if __name__ == "__main__":
+
+  if "DEEPCHEM_DATA_DIR" in os.environ:
+    data_dir = os.environ["DEEPCHEM_DATA_DIR"]
+  else:
+    data_dir = "/tmp"
+
+  save_dir = os.path.join(data_dir, "roitberg")
+
+  loaded, all_datasets, transformers = dc.utils.save.load_dataset_from_disk(
+      save_dir)
+
+  if loaded:
+    train_dataset, valid_dataset, test_dataset = all_datasets
+  else:
+    train_dataset, valid_dataset, test_dataset, transformers = load_roiterberg_ANI(
+    )
+    print("Saving to disk...", save_dir)
+    dc.utils.save.save_dataset_to_disk(save_dir, train_dataset, valid_dataset,
+                                       test_dataset, transformers)
+
+  max_atoms = 23
+  batch_size = 256  # CHANGED FROM 16
+  layer_structures = [128, 128, 64]
+  atom_number_cases = [1, 6, 7, 8, 16]
+
+  metric = [
+      dc.metrics.Metric(dc.metrics.mean_absolute_error, mode="regression"),
+      dc.metrics.Metric(dc.metrics.pearson_r2_score, mode="regression")
+  ]
+
+  n_feat = dc.trans.ANITransformer(
+      max_atoms=max_atoms, atom_cases=atom_number_cases).get_num_feats() - 1
+
+  model = dc.models.ANIRegression(
+      1,
+      max_atoms,
+      n_feat,
+      layer_structures=layer_structures,
+      atom_number_cases=atom_number_cases,
+      batch_size=batch_size,
+      learning_rate=0.001,
+      use_queue=True,
```

Comment:
@miaecle originally use_queue was set to False in your example qm7_ANI.py code, what was the reason?

Reply:
Hi! Sorry for the late reply, setting to False is for the convenience of debugging when I start writing it. Actually I haven't tried using queue in this model, it will be great if queue works here.